src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/ValidateResourceIdFilterAttribute.cs

```diff
@@ -24,19 +25,20 @@ public override void OnActionExecuting(ActionExecutingContext context)
             if (context.RouteData.Values.TryGetValue(KnownActionParameterNames.Id, out var actionId) &&
                 context.ActionArguments.TryGetValue(KnownActionParameterNames.Resource, out var parsedModel))
             {
-                if (!string.Equals((string)actionId, ((Resource)parsedModel).Id, StringComparison.Ordinal))
+                string location = string.Concat(parsedModel.GetType().ToString().Split('.').Last(), ".id");
```

Comment:
You might be able to do `context.RouteData.Values.TryGetValue(KnownActionParameterNames.ResourceType, out var resourceType)` to get the respource type.